### PR TITLE
Migrate from Axios to fetch

### PR DESCRIPTION
**Pull request by vahnag**
## Description of the change

This is pretty much still work-in-progress as there are some rough edges with the code quality that I'm going to improve before it's ready to be merged. 

But I'd like to hear your thoughts about the overall solution.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

https://linear.app/trycourier/issue/C-7120/courier-node-migrate-from-axios-to-native-fetch

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,27 +11,27 @@ const DEFAULTS = {
   BASE_URL: "https://api.courier.com"
 };
 
+const getEnvVariable = (name: string) => globalThis?.process?.env?.[name];
```
Comment (tk26) [inline src/index.ts:14]: do we need to change this with this PR? Unless its absolutely necessary I'd suggest keeping it isolated from axios -> fetch changes?

File: src/audiences/index.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ const putAudience = (options: ICourierClientConfiguration) => {
   return async (
     audience: Omit<AudienceTypes.IAudience, "created_at" | "updated_at">
   ): Promise<AudienceTypes.IAudience> => {
-    const response = await options.httpClient.get<{
+    const response = await options.httpClient.put<{
```
Comment (tk26) [inline src/audiences/index.ts:43]: good catch! cc @suhasdeshpande

File: src/index.ts (hunk 1 above)
Comment (vahnag) [reply src/index.ts:14]: This is necessary to get it to run on Cloudflare workers as there's the `process` global variable there.

**Review by tk26 — APPROVED** [review]

File: src/brands.ts [hunk 3]
```diff
@@ -34,17 +33,10 @@ export const createBrand = (options: ICourierClientConfiguration) => {
...
-    };
-
-    if (config && config.idempotencyKey) {
-      axiosConfig.headers["Idempotency-Key"] = config.idempotencyKey;
-    }
     const res = await options.httpClient.post<ICourierBrand>(
       `/brands`,
       params,
-      axiosConfig
+      { idempotencyKey: config?.idempotencyKey }
```
Comment (scarney81) [inline src/brands.ts:39]: Why are we not supporting expiry here?

Comment (scarney81) [reply src/brands.ts:39]: seems like a gap

Comment (tk26) [reply src/brands.ts:39]: I'll add it as a followup PR

**Review by tk26 — APPROVED** [review]

File: src/http-client.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import crossFetch from "cross-fetch";
+
+import { HttpMethodClient, IHttpClient, IInitHttpClientOptions } from "./types";
+
+const fetch = globalThis.fetch ?? crossFetch;
```
Comment (vahnag) [inline src/http-client.ts:5]: This [issue](https://github.com/lquixada/cross-fetch/issues/69) prevents cross-fetch to fall back to the existing global fetch implementation in Cloudflare workers, so I'm doing it manually.

**Review by tk26 — APPROVED** [review]

**Pull request merged**